demo/predictions/css/legend.css

```diff
@@ -0,0 +1,47 @@
+fieldset {
+    border: 1px solid#ddd;
+    padding: 5px 12px;
+}
+
+legend {
+    font-size: 18px;
+    padding: 0 10px;
+    margin: auto;
+}
+
+.state-predicted-on-time .legend li:first-child::before {
+    background-color: var(--color-state-predicted-on-time);
```

Comment:
**suggestion:** This question is similar to the one about `--color-state-predicted-late`, but in this case, the values for `--color-state-predicted-on-time` and `--color-path-predicted-on-time` are different. 
As a result, I am uncertain about which color to include in the legend, which is quite frustrating. 🙁

Reply:
I agree, I don't remember why we didn't choose to use the same value, this is weird.
I let you choose the color 😄